### PR TITLE
Use SingleHeader format for unspecified B3 inject encoding

### PR DESCRIPTION
**Pull request by gunturaf**
**Which problem is this PR solving?**

This PR changes default behavior when B3 inject encoding is unspecified from multi-header to single-header, per #613.

**Short description of the changes**

- Use single-header format when B3 inject encoding is unspecified
- Modify test data for cases involving unspecified B3 inject encoding

### CODE REVIEW

File: propagators/b3/b3_data_test.go [hunk 1]
```diff
@@ -541,9 +541,7 @@ var injectHeader = []injectTest{
 			TraceFlags: trace.FlagsSampled,
 		},
 		wantHeaders: map[string]string{
-			b3TraceID: traceIDStr,
-			b3SpanID:  spanIDStr,
-			b3Sampled: "1",
+			b3Context: fmt.Sprintf("%s-%s-%s", traceIDStr, spanIDStr, "1"),
 		},
 		doNotWantHeaders: []string{
 			b3ParentSpanID,
```
Comment (MrAlias) [inline propagators/b3/b3_data_test.go:547]: Ensure this is single header now as well.

```suggestion
			b3ParentSpanID,
			b3TraceID,
			b3SpanID,
			b3Sampled,
```

File: propagators/b3/b3_data_test.go [hunk 2]
```diff
@@ -558,9 +556,7 @@ var injectHeader = []injectTest{
 			SpanID:  spanID,
 		},
 		wantHeaders: map[string]string{
-			b3TraceID: traceIDStr,
-			b3SpanID:  spanIDStr,
-			b3Sampled: "0",
+			b3Context: fmt.Sprintf("%s-%s-%s", traceIDStr, spanIDStr, "0"),
 		},
 		doNotWantHeaders: []string{
 			b3ParentSpanID,
```
Comment (MrAlias) [inline propagators/b3/b3_data_test.go:562]:
```suggestion
			b3ParentSpanID,
			b3TraceID,
			b3SpanID,
			b3Sampled,
```

File: propagators/b3/b3_data_test.go [hunk 3]
```diff
@@ -576,8 +572,7 @@ var injectHeader = []injectTest{
 			TraceFlags: trace.FlagsDeferred,
 		},
 		wantHeaders: map[string]string{
-			b3TraceID: traceIDStr,
-			b3SpanID:  spanIDStr,
+			b3Context: fmt.Sprintf("%s-%s", traceIDStr, spanIDStr),
 		},
 		doNotWantHeaders: []string{
 			b3Sampled,
```
Comment (MrAlias) [inline propagators/b3/b3_data_test.go:578]:
```suggestion
			b3Sampled,
			b3TraceID,
			b3SpanID,
```

File: propagators/b3/b3_data_test.go [hunk 4]
```diff
@@ -610,9 +605,7 @@ var injectHeader = []injectTest{
 			TraceFlags: trace.FlagsDebug,
 		},
 		wantHeaders: map[string]string{
-			b3TraceID: traceIDStr,
-			b3SpanID:  spanIDStr,
-			b3Flags:   "1",
+			b3Context: fmt.Sprintf("%s-%s-%s", traceIDStr, spanIDStr, "d"),
 		},
 		doNotWantHeaders: []string{
 			b3Sampled,
```
Comment (MrAlias) [inline propagators/b3/b3_data_test.go:611]:
```suggestion
			b3Sampled,
			traceIDStr,
			spanIDStr,
```

File: propagators/b3/b3_data_test.go [hunk 5]
```diff
@@ -628,9 +621,7 @@ var injectHeader = []injectTest{
 			TraceFlags: trace.FlagsSampled | trace.FlagsDebug,
 		},
 		wantHeaders: map[string]string{
-			b3TraceID: traceIDStr,
-			b3SpanID:  spanIDStr,
-			b3Flags:   "1",
+			b3Context: fmt.Sprintf("%s-%s-%s", traceIDStr, spanIDStr, "d"),
 		},
 		doNotWantHeaders: []string{
 			b3Sampled,
```
Comment (MrAlias) [inline propagators/b3/b3_data_test.go:627]:
```suggestion
			b3Sampled,
			traceIDStr,
			spanIDStr,
			b3Flags,
```

**Review by MrAlias — CHANGES_REQUESTED** [review]

**Review by MrAlias — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**